### DISCUSSION TITLE
Tauri app and missing environment

### DISCUSSION THREAD

**Discussion by VPKSoft** [sushinpv/react-secure-storage#24] [Q&A] [answered]
Hi,
I have a [Tauri Vite](https://tauri.app/v1/guides/getting-started/setup/vite) and after importing the library and referencing it without using, I get the following error: 
```
envHelper.js:20  Uncaught ReferenceError: process is not defined
    at Object.getEnvValue2 [as getEnvValue] (envHelper.js:20:3)
    at getSecurePrefix2 (utils.js:17:39)
    at node_modules/react-secure-storage/dist/localStorageHelpers.js (localStorageHelpers.js:26:29)
    at __require2 (chunk-TFWDKVI3.js?v=d285ff44:18:50)
    at node_modules/react-secure-storage/dist/index.js (index.js:10:51)
    at __require2 (chunk-TFWDKVI3.js?v=d285ff44:18:50)
    at index.js:121:19
```

The reference which is not used is a function:
```typescript
const saveFilePassword = (password: string) => {
    secureLocalStorage.setItem("filePassword", password);
};
```

Might just be that I don't have the `.env` file specified, but thought to ask anyway 🙂
![image](https://user-images.githubusercontent.com/40712699/235320635-29e313a8-f2dc-4a5d-8137-e8e9e48d4b70.png)

So where should I start to get this working?

**Comment by sushinpv** [comment, marked as answer]
The issue is because, the process is not defined in the vite by default,

You can define it by using `vite.config.ts`

    import { defineConfig } from 'vite'
    // ...
    export default defineConfig({
      // ...
      define: {
        'process.env': {}
      }
    }) 
    
 You can find more information here https://github.com/vitejs/vite/issues/1973

**Reply by VPKSoft** [reply]
That did the job! Thank you 👍 I should have probably realized to have searched the issue as Vite-related but didn't realize that at the moment.
